src/uniswap-summaries.md

```diff
@@ -3253,6 +3253,40 @@ module SOLIDITY-UNISWAP-GETRESERVES-SUMMARY
 endmodule
 ```
 
+```k
+module SOLIDITY-MATHSQRT-SUMMARY
+  imports SOLIDITY-CONFIGURATION
+  imports SOLIDITY-UNISWAP-TOKENS
+  imports SOLIDITY-EXPRESSION
+  imports SOLIDITY-STATEMENT
+
+  rule <k> mathSqrt:Id ( v(V:MInt{256}, uint256), .TypedVals ) ~> K => Ss ~> restoreEnv( (y |-> var(size(STORE), uint256)) (z |-> var(size(STORE) +Int 1, uint256)) ) ~> .Statements ~> return z ; ~> .K </k>
+       <summarize> true </summarize>
+       <this-type> TYPE </this-type>
+       <contract-id> TYPE </contract-id>
+       <contract-fn-id> mathSqrt </contract-fn-id>
+       <contract-fn-body> if ( y > 3 ) { z = y ;  uint256 x = y / 2 + 1 ; Ss:Statements } else if ( y != 0 ) { _:Statements }  .Statements </contract-fn-body>
+       <env> E => (y |-> var(size(STORE), uint256))
+                  (z |-> var(size(STORE) +Int 1, uint256))
+                  (x |-> var(size(STORE) +Int 2, uint256))
+       </env>
+       <store> STORE =>
+               STORE ListItem(V) // y
+                     ListItem(V) // z
+                     ListItem((V /uMInt 2p256) +MInt 1p256) // x
+        </store>
+       <current-function> FUNC => mathSqrt </current-function>
+       <call-stack>... .List => ListItem(frame(K, E, FUNC)) </call-stack>
+    requires V >uMInt 3p256 [priority(40)]
+
+  rule <k> mathSqrt:Id ( v(V:MInt{256}, uint256), .TypedVals ) => v (1p256, uint256) ...</k>
```

Comment:
Is there a reason why the cells: `<this-type>`,  `<contract-id>`, `<contract-fn-id>`, etc. wouldn't be necessary here?

Reply:
Yes. This rule replaces the entire call. So, we do not need to try and find the code of the function, we simply return a TypedVal. We know that a `<contract-fn>` with `<contract-fn-id> mathSqrt </contract-fn-id>` exists, but we do not need to find any information from the cells in that `<contract-fn>`.

Reply:
You can think of it just like any other summary of en entire call (which it is with these conditions), e.g.
https://github.com/Pi-Squared-Inc/solidity-demo-semantics/blob/c94634ebc4f33417bd60dc731cee3320186be087/src/uniswap-summaries.md?plain=1#L2658